### DISCUSSION TITLE
Fill in the middle (or in between)

### DISCUSSION THREAD

**Discussion by yvos** [rxhanson/RectanglePro-Community#591] [Ideas]
It would be nice to have a feature called something like Fill in the middle (or in between). It behaves kind like Fill left/right.

Assume the following starting position:
	- window 1 at the left of the screen with 1/3 and height 1
	- window 1 at the right of the screen with 1/25 and height 1
Using Fill in the middle (or in between) fills in the empty space between the above windows. 

_Btw is there a place to see the backlog? And maybe vote..._

**Comment by rxhanson** [comment]
Thanks for the feedback. This one's been on my mind as well. I was also thinking of a type of positioning that would fill starting with the edge of an existing window up to a certain size, so that you could effectively use one shortcut to "stack and fill" windows across a display (hopefully that makes sense :)

> Btw is there a place to see the backlog? And maybe vote...

At the moment, there is not. I've given this a lot of thought (there are pros and cons) and I might add both the visible backlog and voting at some point in the future.

**Reply by yvos** [reply]
Thanks for your response! Although you lost me with the stack and fill idea 🙃

btw I am a new user and it took me quite some time to figure out how powerful the pro version is. Reading through GitHub help me a lot. But man I am happy with it now. Multi monitor setup would be on my whishlist also. Although I fixed this with relative sizes which should work.

**Reply by rxhanson** [reply]
I'll add your +1 to the multi-monitor setup. I think the stack & fill might have to be something that's demoed in order to accurately convey my thoughts :)

**Reply by yvos** [reply]
Thanks for the +1. If you want some feedback on your thoughts / demo just let me know.

**Reply by yvos** [reply]
Hi, just wondering if there’s any update on the “fill in the middle” feature?  

Also, I’m curious if there are any plans to add support for automatic workspace recognition—similar to how Stay can restore window sizes and placements based on the connected monitor setup. What I’m looking for is the ability for Rectangle Pro to automatically detect when I’m at, for example, my home office or a different desk with another set of monitors, and then restore all my windows to their preferred positions without manual intervention.  

Is this kind of feature on the roadmap? Would love to hear your thoughts

Thanks!

**Reply by rxhanson** [reply]
Thanks for checking back in! No updates on "fill in the middle" at this time, although this is still something I'd like to add at some point. 

As for automatic workspace recognition, I do have a hidden toggle for something like this but haven't received any feedback from anyone on whether or not it works for them. Would you mind trying it out if it fits your use case? The idea is that with that single setting enabled, it will simply place windows back to where you had them last when that display was connected.

Execute this Terminal command and restart the app:

```
defaults write com.knollsoft.Hookshot autoSaveRestoreDisplayLayout -int 1
```

Or are you looking more for something like saving an App Layout and specifying when you connect a specific display to apply it?

**Reply by yvos** [reply]
I’ll definitely give the hidden toggle a try and can already see the added value it brings!

As for my ideal feature: I would love to be able to set a unique app layout for each display setup without having to remember a separate shortcut. So for example I have an app layout for when I am in a MS Teams call. In my setup at work I want teams on my 3rd screen in full screen but at home I want is on my 2nd screen on the top right and not full screen.

**Reply by yvos** [reply]
By the way, I did a quick test, it worked perfectly three times out of four so far. I’ll give it some more time and try to replicate the one time it didn’t work well and let you know.

**Reply by yvos** [reply]
Did some more testing and it works perfect. I think the glitch was due to switching displays to fast. In normal usage I don't think it's an issue. I would say that this features deserves a place in the GUI ;-)

**Reply by rxhanson** [reply]
Awesome, thanks for testing that out and letting me know!!

**Reply by yvos** [reply]
Hi,

I have the feeling that something broke in a recent release. I have set: 

`defaults write com.knollsoft.Hookshot autoSaveRestoreDisplayLayout -int 1`

But my windows keep coming up full screen as soon as I connect the display. It's also pretty reproducible, so if you want me to try out and do some tests, it's definitely possible.

**Reply by rxhanson** [reply]
I loaded your config, and your "Fullscreen" app layout is set to be triggered when the display is connected.

**Reply by yvos** [reply]
Oh damn, now I feel really stupid. Sorry to bother you with this.

**Reply by rxhanson** [reply]
No worries! I'm always glad to figure out what's going on :)

**Reply by yvos** [reply]
Hi, did you give the idea for: fill in the middle (or in between) or stack and fill another thought? (see first post)
I would still be very happy with it.

**Reply by rxhanson** [reply]
Been a while since I’ve give this some thought, but I think it makes sense to at least add a fill middle action as a start. I have some adjustments that need to be made to the fill actions and will plan for it at that time. Thanks for following up!
